pkg/kn/commands/service_delete.go

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2018 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewServiceDeleteCommand represent 'service delete' command
+func NewServiceDeleteCommand(p *KnParams) *cobra.Command {
+
+	serviceDeleteCommand := &cobra.Command{
+		Use:     "delete <SERVICE_NAME>",
+		Aliases: []string{"del", "d"},
+		Short:   "Delete a service",
+		Example: `
+  # Delete a service 'svc1' in default namespace
+  kn service delete svc1
+
+  # Delete a service 'svc2' in 'ns1' namespace
+  kn service delete svc2 -n ns1`,
+		// check for arg length == 1
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
```

Comment:
let's move arg check for now into RunE. it seems that we have for now stayed with single RunE configuration in all other commands (imho a good thing since it keeps flow more or less linear).

Reply:
@cppforlife: sounds good. Updated the PR.